ingredient/ingredient-utils/src/functions.ts

```diff
@@ -92,10 +92,20 @@ export class CoreUtils extends BaseUtility {
         return await this.context.Ingredient.properties.source.valueAsync(this.context)
     }
 
-    public get_event_hub_namespace_name(pkgName: string, resourceGroup: string | null = null): string {        
+    public get_event_hub_namespace_name(pkgName: string): string {        
         const evenhHubNamespace = this.create_resource_name("ehn", pkgName, true);
        
         return `${evenhHubNamespace}`;
     }
+
+    public get_app_insights_name(name: string): string {
+        const appInsightsName = this.create_resource_name("ai", name, false);
+
+        return `${appInsightsName}`;
+    }
+
+    public get_app_insights_resource_group(name: string): string {
+        return this.create_resource_name("", name, false);
```

Comment:
This isn't going to do what you want.

This is going to give you a different resource_name depending on the recipe it's run from.

You need to build a stable RG name no matter what recipe executes it so that we can get back to it from everywhere

Look at create_resource_name and you'll see we use the package shortname (and region) to build it.

Reply:
Sorry I'm way behind on this.

Reply:
I moved this util function to the App Insights ingredient.  I made it required for the same reason as the resource group but would appreciate your thoughts.